### PR TITLE
feat: add chainId config to useContractWrite and useSendTransaction

### PR DESCRIPTION
**Pull request by tmm**
## Description

Adds `chainId` configuration parameter to `useContractWrite` and `useSendTransaction`. If `chainId` is provided, the connector will validate that `chainId` is the active chain before sending a transaction (and switch to `chainId` if necessary).

## Additional Information

- [x] I read the [contributing docs](/tmm/wagmi/blob/main/.github/CONTRIBUTING.md) (if this is your first contribution)

Your ENS/address: awkweb.eth

### CODE REVIEW

**Review by jxom — APPROVED** [review]

**Comment by holic** [comment]
😍

File: examples/_dev/src/components/Account.tsx [hunk 1]
```diff
@@ -18,8 +18,9 @@ export const Account = () => {
   })
   const ensAvatar = useEnsAvatar({
     addressOrName: account?.address,
+    chainId: 1,
```
Comment (holic) [inline examples/_dev/src/components/Account.tsx:21]: Should this just be the default for ENS queries? I feel like you more often want the ENS data on mainnet. It's rare that I see someone with an e.g. rinkeby ENS set up properly.

File: packages/core/src/actions/accounts/switchNetwork.ts [hunk 2]
```diff
@@ -11,10 +14,12 @@ export type SwitchNetworkResult = Chain
 export async function switchNetwork({
   chainId,
 }: SwitchNetworkArgs): Promise<SwitchNetworkResult> {
-  const client = getClient()
-  const connector = client.connector
+  const { connector } = getClient()
+  if (!connector) throw new ConnectorNotFoundError()
   if (!connector?.switchChain)
```
Comment (holic) [inline packages/core/src/actions/accounts/switchNetwork.ts:19]: nit: don't need the optional chaining here anymore (`?.`)

File: examples/_dev/src/components/Account.tsx (hunk 1 above)
Comment (tmm) [reply examples/_dev/src/components/Account.tsx:21]: Could make sense. RainbowKit might be heading [in that direction](https://github.com/rainbow-me/rainbowkit/pull/496). Should think about if there are any arguments against this and address in a separate PR.

**Pull request merged**